### PR TITLE
[#IC-301]  Add Event Hub trigger to update cosmos message_view

### PR DESCRIPTION
**Pull request by fabriziopapi**
This function wil keep updated the _message_view_ cosmos collection with the data coming from _message_status_ , _messages_ and _message_content_, merging those data source in a unique collection. With this materialized collection, we could query messages data in a single place.

#### List of Changes

add update message_view function
add appinsight client
update io-functions-commons library

#### Motivation and Context

With the materialized _message_view_ we could query the data directly on cosmos collection, increasing messages pagination and filter performance.   

#### How Has This Been Tested?

unit tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: UpdateCosmosMessageView/handler.ts [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+export const storeAndLogError = <T>(
+  queueClient: QueueClient,
+  telemetryClient: TelemetryClient
+) => (processingError: IStorableError<T>): TE.TaskEither<void, void> =>
+  pipe(
+    processingError,
+    storeError(queueClient),
+    TE.mapLeft(storingError =>
+      telemetryClient.trackEvent({
+        name: "trigger.elt.updatemessageview.failedwithoutstoringerror",
```
Comment (AleDore) [inline UpdateCosmosMessageView/handler.ts:82]: elt?

File: UpdateCosmosMessageView/handler.ts [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        properties: {
+          processingError: JSON.stringify(processingError),
+          storingError: storingError.message
+        },
+        tagOverrides: { samplingEnabled: "false" }
+      })
+    ),
+    TE.map(() =>
+      telemetryClient.trackEvent({
+        name: "trigger.elt.updatemessageview.failed",
```
Comment (AleDore) [inline UpdateCosmosMessageView/handler.ts:92]: ditto

File: UpdateCosmosMessageView/handler.ts [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+  queueClient: QueueClient,
+  blobService: BlobService,
+  rawMessageStatus: unknown
+): Promise<IStorableError<unknown> | MessageView> =>
+  pipe(
+    rawMessageStatus,
+    RetrievedMessageStatusWithFiscalCode.decode,
+    TE.fromEither,
+    TE.chain(handleStatusChange(messageViewModel, messageModel, blobService)),
+    TE.mapLeft(toStorableError(rawMessageStatus)),
```
Comment (AleDore) [inline UpdateCosmosMessageView/handler.ts:213]: I think it should be explicit in the handle status change pipe so we can map directly failures, indicating for each if it's retriable or not and the related type

File: UpdateCosmosMessageView/handler.ts [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    TE.orElseFirst(storeAndLogErrorFirst(queueClient, telemetryClient)),
```
Comment (AleDore) [inline UpdateCosmosMessageView/handler.ts:214]: why this?

File: UpdateCosmosMessageView/handler.ts (hunk 1 above)
Comment (fabriziopapi) [reply UpdateCosmosMessageView/handler.ts:82]: fixed, thank you!

File: UpdateCosmosMessageView/handler.ts (hunk 4 above)
Comment (fabriziopapi) [reply UpdateCosmosMessageView/handler.ts:215]: this is the call to publish the error on the storage queue and track the event on app ins.
The use of _First_ version is to keep the original error as handle return.

File: UpdateCosmosMessageView/handler.ts (hunk 3 above)
Comment (fabriziopapi) [reply UpdateCosmosMessageView/handler.ts:214]: in which case the retriable value could not be mapped from the error type?

File: UpdateCosmosMessageView/handler.ts [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+  pipe(
+    messageViewModel.patch(
+      [messageStatus.messageId, messageStatus.fiscalCode],
+      {
+        status: {
+          archived: messageStatus.isArchived,
+          processing: messageStatus.status,
+          read: messageStatus.isRead
+        }
+      }
```
Comment (AleDore) [inline UpdateCosmosMessageView/handler.ts:128]: Maybe we have to increment version here :)

Comment (fabriziopapi) [reply UpdateCosmosMessageView/handler.ts]: you are right! fixed, thank you!

**Comment by AleDore** [comment]
@fabriziopapi : it seems unit tests fail :)

**Comment by fabriziopapi** [comment]
> @fabriziopapi : it seems unit tests fail :)

The pipeline try to execute the _generate_ comman, that is missing in our package.json.
We must decide if add an empty generation to project or if remove this step from pipeline.

**Pull request closed without merging**